ServerCore/Pages/Hints/AuthorIndex.cshtml.cs

```diff
@@ -0,0 +1,166 @@
+
+        public Team  Team { get; set; }
+
+        public SortOrder? Sort { get; set; }
+
+        public const SortOrder DefaultSort = SortOrder.TimeDescending;
+
+        public async Task<IActionResult> OnGetAsync(int? puzzleId, int? teamId, SortOrder? sort)
+        {
+            Sort = sort;
+
+            if (puzzleId == null)
+            {
+                if (EventRole == EventRole.admin)
+                {
+                    if (teamId == null)
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.Team.Event == Event).ToListAsync();
+                    }
+                    else
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.TeamID == teamId).ToListAsync();
+                    }
+                }
+                else
+                {
+                    // Surely there is a way to get a join to do a bunch of this work, but joins are simply not for me. Someone else can fix later.
+                    var hintStatePerTeam = new List<HintStatePerTeam>();
+                    var authorPuzzles = await UserEventHelper.GetPuzzlesForAuthorAndEvent(_context, Event, LoggedInUser).ToListAsync();
+
+                    if (teamId == null)
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.Team.Event == Event).ToListAsync();
+                        HintStatePerTeam = HintStatePerTeam.Where((h) => authorPuzzles.Contains(h.Hint.Puzzle)).ToList();
+                    }
+                    else
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.TeamID == teamId).ToListAsync();
+                        HintStatePerTeam = HintStatePerTeam.Where((h) => authorPuzzles.Contains(h.Hint.Puzzle)).ToList();
+                    }
+                }
+            }
+            else
+            {
+                Puzzle = await _context.Puzzles.Where(m => m.ID == puzzleId).FirstOrDefaultAsync();
+
+                if (EventRole == EventRole.author && !await UserEventHelper.IsAuthorOfPuzzle(_context, Puzzle, LoggedInUser))
+                {
+                    return Forbid();
+                }
+
+                if (teamId == null)
+                {
+                    HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.Hint.Puzzle == Puzzle).ToListAsync();
+                }
+                else
+                {
+                    HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.TeamID == teamId && h.Hint.Puzzle == Puzzle).ToListAsync();
+                }
+            }
+
+            if (teamId != null)
+            {
+                Team = await _context.Teams.Where(m => m.ID == teamId).FirstOrDefaultAsync();
```

Comment:
Should this check for non-existent teams?

Reply:
seems unlikely but will fix.